backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/KeyValueHandler.java

```diff
@@ -243,16 +255,27 @@ private void insertKeyValue(String entityId, Handle handle, KeyValue kv) {
                     jdbi.useHandle(
                         handle -> {
                           if (exists) {
+                            String value = ModelDBUtils.getStringFromProtoObject(kv.getValue());
+                            // Logic to convert canonical number to double number
+                            if (kv.getValue().hasNumberValue()) {
```

Comment:
Can this logic be pulled into a function so it can be reused here and in `KeyValueEntity` line 37? The two places have very similar logic with only a small change, so if possible it would be nice to not repeat i t. It might also be useful in the future.